lib/datadog/core/remote/client.rb

```diff
@@ -117,7 +117,17 @@ def payload
             tracer_version: Core::Environment::Identity.tracer_version,
             service: Datadog.configuration.service,
             env: Datadog.configuration.env,
-            tags: [], # TODO: add nice tags!
+            tags: [
+              format('ruby.tracer.version:%s', Core::Environment::Identity.tracer_version),
+              format('ruby.runtime.platform:%s', RUBY_PLATFORM),
+              format('ruby.runtime.version:%s', RUBY_VERSION),
+              format('ruby.runtime.engine.name:%s', RUBY_ENGINE),
+              format(
+                'ruby.runtime.engine.version:%s',
+                (defined?(RUBY_ENGINE_VERSION) ? RUBY_ENGINE_VERSION : RUBY_VERSION)
```

Comment:
Would it make sense to use:
```suggestion
                Datadog::Core::Environment::Ext::ENGINE_VERSION
```

Reply:
I'm not too sure, I thought about it but it feels very weird overall: these constants are kind of all over the place and there are helper methods in e.g `core/environment/identity`. It feels like this area is overdue for a refactoring.

I was hoping to punt and hope we'd drop 2.1 and 2.2 soon enough.

Reply:
Also I seem to recall these constants referencing constants referencing constants being returned by methods made some tests gnarly.

I understand the performance angle, but I was thinking maybe a solution would be to memoize behind methods in a dedicated instance. Sure a method call is a tiny bit slower than constant resolution, but then again we're calling `Core::Environment::Identity.lang*` (which I feel is weirdly designed and named, so, larger refactoring in perspective)